Devel - Completed Wok on both #92 and #100

mgmonteleone opened the pull request:
## Issue #92 

Work for issue #92, added both the get all databases as well as get metrics for database.

Added the database metrics to the Measurements class as well.

Currently only getting all metrics for database is supported. if there is need to pull individual metrics, perhaps we can add that later.

--- 

##  Issue #100
As part of completing #100 needed to refactor quite a bit with measurements, including:

- grabbing the unit description from the API
- Changing the _friendly method to use the unit description to properly render bytes of numbers.
- Moved measurements classes to its own module (`measurements.py`)
- had to refactor tests

Commit:
Added endpoints for #100

Commit:
added get_partitions function for #100

Commit:
Moved measurement code out of specs and into own module.

Commit:
Added units to measurements as part of #100

Commit:
Fixed tests as part of #100

Commit:
Added Classes for new disk based measurements for #100

Commit:
Comleted #100, including tests and inline docs.

Commit:
Fixing import issue for #111

Commit:
Added settings (URLs) for #92

Commit:
Added the Namespaces measurement class for #92

Commit:
Completed this list datbases endpoint for #92, including test.

Commit:
Completed get database metrics for #92.

Commit:
Prepared for 2.0.rc1 release,

dmalik-mdb commented on atlasapi/measurements.py:
```diff
@@ -0,0 +1,422 @@
+        except StatisticsError:
+            logger.warning('Could not compute statistical values.')
+            self.samples: int = 0
+            self.mean: int = 0
```
mean as a float for consistency with the declaration above ^^

mgmonteleone replied:
Fixed! New commit coming.

dmalik-mdb commented on atlasapi/settings.py:
```diff
@@ -156,7 +164,7 @@ class Settings:
-    itemsPerPage: int  = int(os.getenv('ITEMS_PER_PAGE', 500))
+    itemsPerPage: int = int(os.getenv('ITEMS_PER_PAGE', 500))
     itemsPerPageMin: int = int(os.getenv('ITEMS_PER_PAGE_MIN', 1))
     itemsPerPageMax: int = int(os.getenv('ITEMS_PER_PAGE_MAX', 2000))
```
Not part of this change but I believe generally we want the max to default to 500, that seems to be the max supported by the clusters and projects endpoints: https://www.mongodb.com/docs/atlas/reference/api/clusters-get-all/

mgmonteleone replied:
I have not experimented with this yet, but I think we can live this for now, as we will be refactoring the entire paging subsystem.

dmalik-mdb commented on atlasapi/specs.py:
```diff
@@ -622,6 +251,172 @@ def add_log_file(self, name: AtlasLogNames, file: BinaryIO) -> None:
+        """
+        if period is None:
+            period = AtlasPeriods.WEEKS_1
+        logger.info(f'The granularity is {granularity}')
```
fix message to log period rather than granularity

mgmonteleone replied:
Thanks! Fixed. Will push new commit.

dmalik-mdb commented on atlasapi/specs.py:
```diff
@@ -622,6 +251,172 @@ def add_log_file(self, name: AtlasLogNames, file: BinaryIO) -> None:
+        """
+        if period is None:
+            period = AtlasPeriods.WEEKS_1
+        logger.info(f'The granularity is {granularity}')
```
same as above, change log message to print the period

mgmonteleone replied:
Fixed here too, new commit coming!

dmalik-mdb commented on atlasapi/measurements.py:
```diff
@@ -0,0 +1,422 @@
+            data_type: The datatype either bytes or number
+        """
+        try:
+            if data_type is None:
+                data_type = 'SCALAR_PER_SECOND'
+            if data_type == 'BYTES':
+                self.mean: str = hf.format_size(mean(clean_list(data_list)))
+                self.min: str = hf.format_size(min(clean_list(data_list)))
+                self.max: str = hf.format_size(max(clean_list(data_list)))
+            elif data_type == 'SCALAR':
+                self.mean: str = hf.format_number(mean(clean_list(data_list)))
+                self.min: str = hf.format_number(min(clean_list(data_list)))
+                self.max: str = hf.format_number(max(clean_list(data_list)))
+            else:
+                self.mean: str = hf.format_number(mean(clean_list(data_list)))
+                self.min: str = hf.format_number(min(clean_list(data_list)))
+                self.max: str = hf.format_number(max(clean_list(data_list)))
```
are these the only measurement units we'll see? I can't get a definitive answer from the API documentation but Ops Manager docs seem to imply that theres more options: https://www.mongodb.com/docs/ops-manager/current/reference/api/measures/get-measurement-types/

dmalik-mdb replied:
Nevermind - I totally missed the else there. Should be good to go!

dmalik-mdb reviewed (commented):
Left some comments on minor fixes. The only major thing is whether there are additional measurement units that might be recorded as SCALAR_PER_SECOND in the friendly output. Other than that LGTM and amazing that you knocked this out in a couple of days!

Commit:
Fixed two issues from PR review

#113.

https://github.com/mgmonteleone/python-atlasapi/pull/113#discussion_r902884359

Commit:
Fixed one issues from PR review # 113:

Commit:
PR #113, prepared for 2.0.0 release

dmalik-mdb reviewed (approved):
Looking good! 👍
